src/NSwag.AspNetCore/Middlewares/AspNetCoreSwaggerMiddleware.cs

```diff
@@ -0,0 +1,90 @@
+﻿//-----------------------------------------------------------------------
+// <copyright file="SwaggerMiddleware.cs" company="NSwag">
+//     Copyright (c) Rico Suter. All rights reserved.
+// </copyright>
+// <license>https://github.com/NSwag/NSwag/blob/master/LICENSE.md</license>
+// <author>Rico Suter, mail@rsuter.com</author>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.ApiExplorer;
+using NSwag.SwaggerGeneration;
+using NSwag.SwaggerGeneration.AspNetCore;
+
+namespace NSwag.AspNetCore.Middlewares
+{
+    /// <summary>Generates a Swagger specification on a given path.</summary>
+    public class AspNetCoreSwaggerMiddleware
```

Comment:
Adding a middleware + TestHost seemed like the easiest way to do E2E tests (https://github.com/RSuter/NSwag/issues/1038). I added an Mvc 1.1 application, I plan on adding the 1.0 \ 2.0 variants as a follow up once we can figure out if this is the best approach and where this type needs to live.